scripts/core/editor3/components/Editor3.jsx

```diff
@@ -18,30 +20,51 @@ import Toolbar from './toolbar';
  * @description Editor3 is a draft.js based editor that support customizable
  *  formatting (shortly more features will come ...)
  */
+
 export class Editor3 extends React.Component {
     constructor(props) {
         super(props);
 
         const initialContentState = stateFromHTML(props.value);
 
+        this.focus = this.focus.bind(this);
+        this.onChange = this.onChange.bind(this);
+
+        this.handleKeyCommand = this.handleKeyCommand.bind(this);
+        this.onTab = this.onTab.bind(this);
+
+        this.getDecorators = this.getDecorators.bind(this);
+        this.getEditorState = this.getEditorState.bind(this);
+        this.setEditorState = this.setEditorState.bind(this);
```

Comment:
In React, state is considered internal and components are only allowed to communicate via props. For this reason, state should not be shared, but only values via properties. See this [link](https://github.com/uberVU/react-guide/blob/master/props-vs-state.md#state) for an explanation.

Alternatively, I would suggest using the pattern that I've used with toolbar, and simply share `editorState` and `onChange`.